src/compiler/transformers/map-imports-to-path-aliases.ts

```diff
@@ -21,65 +21,63 @@ export const mapImportsToPathAliases = (
-    let sourceFile: string;
-
     const compilerHost = ts.createCompilerHost(config.tsCompilerOptions);
 
-    const visit = (node: ts.Node): ts.VisitResult<ts.Node> => {
-      // We should only attempt to transpile standard module imports:
-      // - import * as ts from 'typescript';
-      // - import { Foo, Bar } from 'baz';
-      // - import { Foo as Bar } from 'baz';
-      // - import Foo from 'bar';
-      // We should NOT transpile other import declaration types:
-      // - import a = Foo.Bar
-      if (ts.isImportDeclaration(node) && ts.isStringLiteral(node.moduleSpecifier)) {
-        let importPath = node.moduleSpecifier.text;
+    const visit =
+      (sourceFile: string) =>
+      (node: ts.Node): ts.VisitResult<ts.Node> => {
+        // We should only attempt to transpile standard module imports:
+        // - import * as ts from 'typescript';
+        // - import { Foo, Bar } from 'baz';
+        // - import { Foo as Bar } from 'baz';
+        // - import Foo from 'bar';
+        // We should NOT transpile other import declaration types:
+        // - import a = Foo.Bar
+        if (ts.isImportDeclaration(node) && ts.isStringLiteral(node.moduleSpecifier)) {
+          let importPath = node.moduleSpecifier.text;
 
-        // We will ignore transforming any paths that are already relative paths or
-        // imports from external modules/packages
-        if (!importPath.startsWith('.')) {
-          const module = ts.resolveModuleName(importPath, sourceFile, config.tsCompilerOptions, compilerHost);
+          // We will ignore transforming any paths that are already relative paths or
+          // imports from external modules/packages
+          if (!importPath.startsWith('.')) {
+            const module = ts.resolveModuleName(importPath, sourceFile, config.tsCompilerOptions, compilerHost);
 
-          const hasResolvedFileName = module.resolvedModule?.resolvedFileName != null;
-          const isModuleFromNodeModules = module.resolvedModule?.isExternalLibraryImport === true;
-          const shouldTranspileImportPath = hasResolvedFileName && !isModuleFromNodeModules;
+            const hasResolvedFileName = module.resolvedModule?.resolvedFileName != null;
+            const isModuleFromNodeModules = module.resolvedModule?.isExternalLibraryImport === true;
+            const shouldTranspileImportPath = hasResolvedFileName && !isModuleFromNodeModules;
 
-          if (shouldTranspileImportPath) {
-            // Create a global extension matching regular expression
-            // This can be used to strip all file extensions off the generated import path string
-            const extensionRegex = new RegExp(Object.values(ts.Extension).join('|'), 'g');
+            if (shouldTranspileImportPath) {
+              // Create a regular expression that will be used to remove the last file extension
+              // from the import path
+              const extensionRegex = new RegExp(Object.values(ts.Extension).join('$|'));
```

Comment:
Small bug - `join` will only be applied when there are >1 entries, and when it is applied, won't be applied to the last entry. So if `ts.Extension` was `['.js', '.ts', '.mjs']`, this only gives us:
```
new RegExp(Object.values(['.js', '.ts', '.mjs']).join('$|')) // Yields  /.js$|.ts$|.mjs/
```
I think we want something more like:
```
new RegExp(Object.values(['.js', '.ts', '.mjs']).map((ext) => `${ext}$`).join('|')) // Yields /.js$|.ts$|.mjs$/
```

Specifically:
```
new RegExp(Object.values(ts.Extension).map((ext) => `${ext}$`).join('|'))
```

Reply:
Oof yeah you're right, silly mistake on my part. Thanks for the callout!